### PR TITLE
Iterable<CodeBlock>.joinToCode

### PR DESCRIPTION
**Pull request by geralt-encore**
https://github.com/square/kotlinpoet/issues/128

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt [hunk 1]
```diff
@@ -379,3 +379,20 @@ class CodeBlock private constructor(
         = c == '%' || c == '>' || c == '<' || c == '[' || c == ']' || c == 'W'
   }
 }
+
+@JvmOverloads
+fun Iterable<CodeBlock>.joinToCode(separator: CharSequence = ", ", prefix: CharSequence = "",
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: Would something like `this.map { it.toString() }.joinToString()` work?

Comment (geralt-encore) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: It should work. Didn't think about it in a first place...

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt [hunk 2]
```diff
@@ -379,3 +379,8 @@ class CodeBlock private constructor(
...
+                                   suffix: CharSequence = "" ) =
+    CodeBlock.of(map { it.toString() }.joinToString(separator, prefix, suffix))
```
Comment (JakeWharton) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: Calling `toString` prevents the ability to import types in these blocks.

It should be something like
```kotlin
val blocks = toTypedArray<CodeBlock>()
return CodeBlock.of(repeat("%L", blocks.size).joinToString(separator, prefix, suffix), *blocks)
```

Comment (geralt-encore) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: I see. Wasn't sure about it. Thanks for pointing out!

Comment (geralt-encore) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: `toTypedArray` is available for `Collection` but not for `Iterable`. I've changed extension type to `Collection` for now. If it is needed I can do revert it back to `Iterable` and convert to array manually.

**Comment by JakeWharton** [comment]
Merged manually as c1abb0f614705e7ffa5b3b444e1438051dcfb152

**Pull request closed without merging**